server/api/api/endpoints/model_monitoring.py

```diff
@@ -0,0 +1,148 @@
+            project,
+            mlrun.common.schemas.model_monitoring.ProjectSecretKeys.ACCESS_KEY,
+        )
+
+    MonitoringDeployment().deploy_monitoring_functions(
+        project=project,
+        model_monitoring_access_key=model_monitoring_access_key,
+        db_session=db_session,
+        auth_info=auth_info,
+        image=image,
+        base_period=base_period,
+    )
+
+
+@router.post("/model-monitoring-controller")
+async def update_model_monitoring_controller(
+    project: str,
+    auth_info: mlrun.common.schemas.AuthInfo = fastapi.Depends(
+        deps.authenticate_request
+    ),
+    db_session: Session = fastapi.Depends(deps.get_db_session),
+    image: str = "mlrun/mlrun",
+    base_period: int = 10,
+):
+    """
+    Redeploy model monitoring application controller function.
+    The main goal of the controller function is to handle the monitoring processing and triggering applications.
+
+    :param project:                  Project name.
+    :param auth_info:                The auth info of the request.
+    :param db_session:               A session that manages the current dialog with the database.
+    :param image:                    The default image of the model monitoring controller job. Note that the writer
+                                     function, which is a real time nuclio functino, will be deployed with the same
+                                     image. By default, the image is mlrun/mlrun.
+    :param base_period:              The time period in minutes in which the model monitoring controller function
+                                     triggers. By default, the base period is 10 minutes.
+    """
+
+    await server.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+        resource_type=mlrun.common.schemas.AuthorizationResourceTypes.function,
+        project_name=project,
+        resource_name=mlrun.common.schemas.model_monitoring.MonitoringFunctionNames.APPLICATION_CONTROLLER,
+        action=mlrun.common.schemas.AuthorizationAction.store,
+        auth_info=auth_info,
+    )
+
+    model_monitoring_access_key = None
+    if not mlrun.mlconf.is_ce_mode():
+        # Generate V3IO Access Key
+        model_monitoring_access_key = process_model_monitoring_secret(
+            db_session,
+            project,
+            mlrun.common.schemas.model_monitoring.ProjectSecretKeys.ACCESS_KEY,
+        )
+    try:
+        # validate that the model monitoring stream has not yet been deployed
+        mlrun.runtimes.nuclio.function.get_nuclio_deploy_status(
+            name=mlrun.common.schemas.model_monitoring.MonitoringFunctionNames.APPLICATION_CONTROLLER,
+            project=project,
+            tag="",
+            auth_info=auth_info,
+        )
+
+    except mlrun.errors.MLRunNotFoundError:
+        raise mlrun.errors.MLRunNotFoundError(
+            f"{mlrun.common.schemas.model_monitoring.MonitoringFunctionNames.APPLICATION_CONTROLLER} does not exists. "
```

Comment:
```suggestion
            f"{mlrun.common.schemas.model_monitoring.MonitoringFunctionNames.APPLICATION_CONTROLLER} does not exist. "
```
https://github.com/mlrun/mlrun/pull/5178#discussion_r1499176614
🙂